### DISCUSSION TITLE
The trick to find a `<td>` by its corresponding `<th>` position in XPath?

### DISCUSSION THREAD

**Discussion by yajo** [OCA/odoo-community.org#111] [Q&A] [answered]
Hi there guys.

I remember there was a way to do this but I can't remember where I saw it...

The point is I have a `<th name="th_sm_ordered"><strong>Ordered</strong></th>` in a report and I have a `<td>` (down below, inside the `<tbody>`) which has no indication of what column is it targeting.

So, how to get the position of the `<th>` inside the `<thead>` and reuse it to get the appropriate `<td>`? 🤔 

Particularly, I want to find [this `<td>`](https://github.com/odoo/odoo/blob/643141dd284f1da869f540dd7aabbe9753eec3a5/addons/stock/report/report_deliveryslip.xml#L80-L83) based on the index position of [this `<th>`](https://github.com/odoo/odoo/blob/643141dd284f1da869f540dd7aabbe9753eec3a5/addons/stock/report/report_deliveryslip.xml#L67).

**Comment by yajo** [comment, marked as answer]
Ha! Found it!

Well, for the sake of the next one coming to this (including my future self), this was the solution:

```xml
    <template id="report_delivery_document" inherit_id="stock.report_delivery_document">
        <xpath expr="//table[@name='stock_move_table']/tbody//td[
                1 + count(
                    //table[@name='stock_move_table']
                    //th[@name='th_sm_ordered']
                    /preceding-sibling::*
                )
            ]" position="inside">
            <!-- Yay! 🎉 -->
        </xpath>
    </template>
```

Inspiration: https://stackoverflow.com/a/227080/1468388

**Reply by simahawk** [reply]
IMO its easier to look for the specific content and walk up. Eg:

```xml
 <xpath expr="//table[@name='stock_move_table']/tbody//td//span[@t-field='move.product_id']/.."> 
```
or

```xml
 <xpath expr="//table[@name='stock_move_table']/tbody//td//span[@t-field='move.product_id']/ancestor::td"> 
```
